assets/app/scripts/directives/popups.js

```diff
@@ -5,13 +5,44 @@ angular.module('openshiftConsole')
   .directive('toggle', function() {
     return {
       restrict: 'A',
+      scope: {
+        dynamicContent: '@?'
+      },
       link: function($scope, element, attrs) {
         if (attrs) {
           switch(attrs.toggle) {
             case "popover":
+              // If dynamic-content attr is set at all, even if it hasn't evaluated to a value
+              if (attrs.dynamicContent || attrs.dynamicContent === "") {
+                // if we actually have a value
+                if ($scope.dynamicContent) {
```

Comment:
I could be wrong, but I think the `$watch` below fires immediately. You might be able to get rid of this `if` block.